collector/lib/HostInfo.cpp

```diff
@@ -179,6 +182,75 @@ bool HostInfo::HasEBPFSupport() {
   return collector::hasEBPFSupport(kernel, GetOSID());
 }
 
+bool HostInfo::HasBTFKernelSupport() {
+  std::string btf_vmlinux_path = GetHostPath("/sys/kernel/btf/vmlinux");
+
+  if (faccessat(AT_FDCWD, btf_vmlinux_path.c_str(), R_OK, AT_EACCESS) == -1) {
```

Comment:
I used a similar implementation to what can be found in libbpf, because I wasn't sure about the consequences of not using the _effective user and groupo IDs_ (`AT_EACCESS`).